crates/objc2-proc-macros/src/extern_methods.rs

```diff
@@ -2,6 +2,50 @@ use proc_macro2::TokenStream;
 use quote::ToTokens;
 use syn::parse::{Parse, ParseStream};
 
+enum RetainSemantics {
+    Init,
+    New,
+    Other,
+}
+
+// Examples:
+//
+//   #[method(sel = initTextCell:, retain = init)]
+//   pub unsafe fn initTextCell(
+//       this: Option<Allocated<Self>>,
+//       string: &NSString,
+//   ) -> Id<Self, Shared>;
+//
+//   #[method(sel = loadHTMLString:baseURL:, retain)] // retain without = defaults to `other`
```

Comment:
We could probably make specifying even `retain` optional in the typical case since we could just determine this by looking at the return type of the signature.

Reply:
The reason that I did it the `@__retain_semantics` `extern_methods!` originally is because of performance (otherwise it uses `const` stuff to figure the selector out), but doing such work is possibly performant enough when running as a proc macro, so we might not need it at all.